src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1520,51 +1542,41 @@ private String quoteUnixCredentials(String str) {
         return str.replace("'", "'\\''");
     }
 
-    private File createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException {
-        File ssh = File.createTempFile("pass", ".bat");
-        try (PrintWriter w = new PrintWriter(ssh)) {
-            w.println("echo \"" + quoteWindowsCredentials(Secret.toString(sshUser.getPassphrase())) + "\"");
-            w.flush();
-        }
-        ssh.setExecutable(true);
+    private FilePath createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException, InterruptedException {
+        FilePath ssh = workspace.createTextTempFile("pass", ".bat",
```

Comment:
no if this introduces a security vulnerability (by being able to retrive credentials/keys via the workspace browser) the it is not another topic at all.